packages/playground/website/src/lib/tracking.ts

```diff
@@ -1,23 +1,69 @@
+import { StepDefinition } from '@wp-playground/blueprints';
+import { logger } from '@php-wasm/logger';
+
 /**
  * Declare the global window.gtag function
  */
 declare global {
-    interface Window { gtag: any; }
+	interface Window {
+		gtag: any;
+	}
 }
 
 /**
  * Google Analytics event names
  */
-type GAEvent = 'load' | 'step';
+type GAEvent = 'load' | 'step' | 'install' | 'error';
 
 /**
  * Log a tracking event to Google Analytics
  * @param GAEvent The event name
  * @param Object Event data
  */
-export const logTrackingEvent = (event: GAEvent, data?: {[key: string]: string}) => {
-    if (typeof window === 'undefined' || !window.gtag) {
-        return;
-    }
-    window.gtag('event', event, data);
-}
+export const logTrackingEvent = (
+	event: GAEvent,
+	data?: { [key: string]: string }
+) => {
+	try {
+		if (typeof window === 'undefined' || !window.gtag) {
+			return;
+		}
+		window.gtag('event', event, data);
+	} catch (error) {
+		logger.warn('Failed to log tracking event', event, data, error);
+	}
+};
+
+/**
+ * Log Blueprint step events
+ * @param step The Blueprint step
+ */
+export const logBlueprintStepEvent = (step: StepDefinition) => {
+	/**
+	 * Log the names of provided Blueprint's steps.
+	 * Only the names (e.g. "runPhp" or "login") are logged. Step options like
+	 * code, password, URLs are never sent anywhere.
+	 */
+	logTrackingEvent('step', { step: step.step });
+
+	if (step.step === 'installPlugin' && (step as any).pluginData.slug) {
```

Comment:
Does this account for `blueprint.plugins`? Or just `blueprint.steps`?

Reply:
Both. It will run whenever a step is completed. 
During Blueprint compile, `blueprint.plugins` [is converted to steps. ](https://github.com/WordPress/wordpress-playground/blob/85f868602c1800d28a9b5788ee8e0d9bb8113f53/packages/playground/blueprints/src/lib/compile.ts#L143)